ui/components/app/wallet-overview/coin-buttons.tsx

```diff
@@ -276,6 +308,21 @@ const CoinButtons = ({
   ///: END:ONLY_INCLUDE_IF
 
   const handleSendOnClick = useCallback(async () => {
+    trackEvent(
+      {
+        event: MetaMetricsEventName.NavSendButtonClicked,
+        category: MetaMetricsEventCategory.Navigation,
+        properties: {
+          account_type: account.type,
+          token_symbol: nativeToken,
+          location: 'Home',
+          text: 'Send',
+          chain_id: chainId,
+          ...getSnapAccountMetaMetricsPropertiesIfAny(account),
```

Comment:
Can the case where `snap_id` is `undefined` be a problem?

Reply:
You mean like having `internalAccount.metadata.snap?.id` being `undefined`?

I don't think that would be problem, no. Also, we already check if `internalAccount.metadata.snap` is defined, so if its `id` is also `undefined` then we might have another problem IMO 😅

Reply:
I actually went for another implementation, so we should not have any `undefined` at all anymore